### ISSUE TITLE
[QUESTION] How to exclude function argument default values from Variable + MatcherType.String in named function expressions?

### ISSUE THREAD

**Issue by Soundvessel** [schoero/eslint-plugin-readable-tailwind#332]
### Question

I have the following pattern matcher for use with variables that start with 'pt':

```js
        {
          kind: SelectorKind.Variable,
          match: [
            {
              type: MatcherType.String,
            },
            {
              type: MatcherType.ObjectValue,
              path: 'class',
            },
          ],
          name: '^pt[A-Z]{1}[a-zA-Z]*$',
        }
```

The problem is that the string matcher is throwing `no-unknown-class` errors on function argument defaults on my function expressions.

These works correctly:
```js
const ptTest = 'unknown-class'

const ptTestTwo = {
  root: 'not-linted',
  class: 'unknown-class',
}
```

My problem is that I need to exclude argument defaults in named function expressions (Airbnb guideline we follow) like so. Here 'sm' is flagged when I just want the above matches.
```js
const ptIssue = function generatePt({ breakpoint = 'sm'}) { /* ... */ }
```

I feel it would make more sense for Variable matchers to ignore the function value which can be covered by Callee matchers.

As a workaround for now I use an additional const that isn't matched:

```js
const doesNotMatch = function generatePt({ breakpoint = 'sm'}) { /* ... */ }

export const ptMenu = doesNotMatch
```

**Comment by schoero** [comment]
I fixed it in the latest release. I noticed that there are other issues with default assignments that I will resolve in a follow up:

- `ObjectValue` matchers should select default assignments:

  ```ts
  function func(params = { classes: " flex " }) { }
  ```

- `String` matchers should not match values in object patterns

  ```ts
  func({ classes = " flex "})
  ```
